### ISSUE TITLE
[GPU] Missing `PM4_EVENT_WRITE_CFL` / 0x59 opcode, causing XEX1 titles to hang

### ISSUE THREAD

**Issue by emoose** [xenia-project/xenia#2056]
Did some testing with a few XEX1 titles, was able to get some running at https://github.com/emoose/xenia/issues/46

E.g. `Football Manager 2005 (2005-08-08) (beta)`:

![xenia_22-09-06_15-11-02-329](https://user-images.githubusercontent.com/1755499/188658108-d3280466-c2fd-4d5e-86d9-94bafabbbaad.png)

Didn't take too much thankfully, found two different methods that could let it work:
- Change `VdGetGraphicsAsicID` to return 0x10
- Add `PM4_EVENT_WRITE_CFL` to `CommandProcessor::ExecutePacketType3`, running same code as `PM4_EVENT_WRITE_SHD`

The first option probably isn't a good idea since it seems retail kernel always returns 0x11 for it (same as Xenia), the second option does seems to work fine (was used in the pic above), but no idea if Xenia should maybe be implementing it differently to _SHD (though from the dolphin XEX comparison in the link it seems to treat them pretty much the same :shrug; )

Kinda interesting that Xenia never had to implement that already, guess no retail titles ever make use of it, must have been dropped during beta->final for some reason.

**Comment by gibbed** [comment]
Do XEX1 files have version imports like XEX2 files do?

I haven't personally looked at the XEX1 format yet.

**Comment by emoose** [comment]
Yep it's mostly the same as XEX2 besides security info having some fields moved around IIRC, other than that I think all the other headers match up.

eg here's log output for the football manager XEX1:

```
i> 0000A3DC Module \Device\Harddisk0\Partition1\fm.exe:
    Module Flags: 00000001
Security Header:
     Image Flags: 00000000
    Load Address: 82000000
      Image Size: 013B0000
    Export Table: 830DBA8C
Optional Header Count: 11
  XEX_HEADER_FILE_FORMAT_INFO (TODO):
  XEX_HEADER_ORIGINAL_BASE_ADDRESS: 00400000
  XEX_HEADER_ENTRY_POINT: 82908A10
  XEX_HEADER_IMAGE_BASE_ADDRESS: 82000000
  XEX_HEADER_IMPORT_LIBRARIES:
    xam.xex - 142 imports
      Version: 0.0.1838.32
      Min Version: 0.0.1838.32
    xboxkrnl.exe - 310 imports
      Version: 0.0.1838.32
      Min Version: 0.0.1838.32
  Unknown Header 00010400
  XEX_HEADER_CHECKSUM_TIMESTAMP (TODO):
  XEX_HEADER_STATIC_LIBRARIES:
    XRTLLBB  : 2.0.1838.0
    XAPILIB  : 2.0.1838.0
    XBOXKRNL : 2.0.1838.0
    D3D9     : 2.0.1838.0
    D3DX9    : 2.0.1838.0
    XGRAPHC  : 2.0.1838.0
    XNET     : 2.0.1838.0
    XAUD     : 2.0.1838.0
    LIBC     : 2.0.1838.0
    XRTLLBB  : 2.0.1838.0
  XEX_HEADER_TLS_INFO:
          Slot Count: 64
    Raw Data Address: 00000000
           Data Size: 0
       Raw Data Size: 0
  XEX_HEADER_DEFAULT_STACK_SIZE: 262144
  XEX_HEADER_GAME_RATINGS (TODO):
```

There's also some formats even earlier than XEX1 like XEX%, not sure if there's much chance of Xenia supporting those though since the import ordinals in those were a lot different IIRC.

**Comment by emoose** [comment]
Ah darn, looks like import ordinals are different on 1838 too, noticed that call to `XamEnumerate` was going to our `XamContentGetCreator` function ;_;

With xam_table.inc updated with the 1838 ordinals it fixes the crash that `Condemned (2005-08-10)` had, and actually lets us get in-game 😱 

![xenia_22-09-06_21-58-11-378](https://user-images.githubusercontent.com/1755499/188736843-fd0c0a80-35ad-4125-a29c-f53561a967ac.jpg)

Here's xam_table.inc with the 1838 changes: https://gist.github.com/emoose/c12ae9ced0ffa94a8d287e082ff9866c
1838 goes up to 0x000007D4, then I stuffed all the extra exports which 1838 was lacking all as 0x000007D5 so they wouldn't need to all be commented/removed.

Not sure if kernel might also need changes too, will try looking at it soon.
